src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -616,8 +612,8 @@ void Assembler::emit_operand_helper(int reg_enc, int base_enc, int index_enc,
     if (is_valid_encoding(index_enc)) {
       assert(scale != Address::no_scale, "inconsistent address");
       // [base + index*scale + disp]
-      if (disp == 0 && no_relocation &&
-          base_enc != rbp->encoding() LP64_ONLY(&& base_enc != r13->encoding())) {
+      if (disp == 0 && no_relocation && base_enc != rbp->encoding()
+          LP64_ONLY(&& base_enc != r13->encoding() && base_enc != 21 && base_enc != 29)) {
```

Comment:
This could be simplified to:
 if (disp == 0 && no_relocation && ((base_enc & 0x7) != 5)) {

Reply:
Done. Thanks.